platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/reports/processor/AppExitAnrTraceProcessor.kt

```diff
@@ -118,4 +123,53 @@ internal object AppExitAnrTraceProcessor {
             mainStackTraceFrames.add(frame)
         }
     }
+
+    private fun extractAnrReason(description: String?): AnrReason {
+        if (description == null) return Generic
+        val sanitizedDescription = description.lowercase()
+
+        return when {
+            "input dispatching timed out" in sanitizedDescription -> AnrReason.UserPerceivedAnr
+            "broadcast of intent" in sanitizedDescription -> BroadcastReceiver
+            "executing service" in sanitizedDescription -> ExecutingService
+            "service.startforeground() not called" in sanitizedDescription -> StartForegroundNotCalled
+            "content provider timeout" in sanitizedDescription -> AnrReason.ContentProvider
+            "app registered timeout" in sanitizedDescription -> AnrReason.AppRegistered
+            "short fgs timeout" in sanitizedDescription -> AnrReason.ShortFgsTimeout
+            "job service timeout" in sanitizedDescription -> AnrReason.JobService
+            "app start timeout" in sanitizedDescription -> AnrReason.AppStart
+            "service start timeout" in sanitizedDescription -> AnrReason.ServiceStart
+            else -> Generic
+        }
+    }
+
+    /**
+     * Based on android source (see frameworks/base/core/java/com/android/internal/os/TimeoutRecord.java)
+     */
+    private sealed class AnrReason(
+        val message: String,
+    ) {
+        // Note: Combining all Input Dispatching Timed Out as User Perceived ANR as per public definition
+        data object UserPerceivedAnr : AnrReason("User Perceived ANR")
+
+        data object BroadcastReceiver : AnrReason("Broadcast Receiver ANR")
+
+        data object ExecutingService : AnrReason("Executing Service ANR")
+
+        data object StartForegroundNotCalled : AnrReason("Service.startForeground() Not Called ANR")
+
+        data object ContentProvider : AnrReason("Content Provider ANR")
+
+        data object AppRegistered : AnrReason("App Registered ANR")
+
+        data object ShortFgsTimeout : AnrReason("Short Foreground Service Timeout ANR")
+
+        data object JobService : AnrReason("Job Service ANR")
+
+        data object AppStart : AnrReason("App Start ANR")
+
+        data object ServiceStart : AnrReason("Service Start ANR")
+
+        data object Generic : AnrReason("Generic ANR")
```

Comment:
>"Generic ANR"
sounds a little weird without a definition, how about something like "Undetermined ANR" or something like that?